pkg/provisioning/provision.go

```diff
@@ -0,0 +1,240 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package provisioning
+
+import (
+	"context"
+	"io/ioutil"
+
+	"github.com/conduitio/conduit/pkg/connector"
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/conduitio/conduit/pkg/pipeline"
+	"github.com/conduitio/conduit/pkg/processor"
+)
+
+var update = false
+
+type Service struct {
+	ctx              *context.Context
+	logger           log.CtxLogger
+	pipelineService  *pipeline.Service
+	connectorService *connector.Service
+	processorService *processor.Service
+}
+
+func GetService(
+	ctx *context.Context,
+	logger log.CtxLogger,
+	plService *pipeline.Service,
+	connService *connector.Service,
+	procService *processor.Service,
+) *Service {
+	return &Service{
+		ctx:              ctx,
+		logger:           logger.WithComponent("provision"),
+		pipelineService:  plService,
+		connectorService: connService,
+		processorService: procService,
+	}
+}
+
+func (s *Service) ProvisionConfigFile(path string) error {
```

Comment:
So there will be a config file directory that we will scan for yaml files and provision all pipelines in that directory. That directory will be determined when Conduit starts - default is `./pipelines` but can be overridden using a CLI flag.

I'm thinking that it would make sense to pass this directory path to `Service` when creating it in `NewService` and store it in the service. This will enable us to write a hot reload function in the future if we need it, since we will already know the directory path.

I would then expect to see an `Init` function here that scans the folder and provisions the pipelines, this way we would follow the pattern we have in other services (see [this](https://github.com/ConduitIO/conduit/blob/30a574c86c10d9280517a0ea086339dd8bbd5662/pkg/conduit/runtime.go#L207-L218)).

Reply:
Oh ok, if you wanted to split it, that's fine!